contracts/vesting.tmy/src/vesting.tmy.cpp

```diff
@@ -124,11 +123,24 @@ namespace vestingtoken
 
                 // Update the tokens_claimed field
                 eosio::asset tokens_claimed = eosio::asset(claimable, vesting_allocation.tokens_claimed.symbol);
-                vesting_table.modify(iter, get_self(), [&](auto &row)
-                                     { row.tokens_claimed = tokens_claimed; });
+
+                if (claimable == vesting_allocation.tokens_allocated.amount)
+                {
+                    // Erase and update iterator correctly
+                    iter = vesting_table.erase(iter);
+                    break;
```

Comment:
should it break? i think it should continue to the next allocation. does it do that?